indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/expressions/ChoiceExpression.java

```diff
@@ -0,0 +1,87 @@
+package com.yahoo.vespa.indexinglanguage.expressions;
+
+import com.yahoo.document.DataType;
+import com.yahoo.document.datatypes.FieldValue;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * An expression which returns the value of the first of a list of subexpressions which
+ * returns a non-null value.
+ *
+ * Syntax: sub-expression1 || sub-expression2 || ...
+ *
+ * @author bratseth
+ */
+public class ChoiceExpression extends ExpressionList<Expression> {
+
+    public ChoiceExpression() {
+        this(List.of());
+    }
+
+    public ChoiceExpression(Expression... choices) {
+        this(Arrays.asList(choices));
+    }
+
+    public ChoiceExpression(Collection<? extends Expression> choices) {
+        super(choices, resolveInputType(choices));
+    }
+
+    @Override
+    protected void doExecute(ExecutionContext context) {
+        FieldValue input = context.getValue();
+        for (Expression expression : this) {
+            context.setValue(input).execute(expression);
+            if (context.getValue() != null)
+                break; // value found
+        }
+    }
+
+    @Override
+    protected void doVerify(VerificationContext context) {
+        DataType input = context.getValueType();
+        for (Expression exp : this)
+            context.setValueType(input).execute(exp);
+        context.setValueType(input);
+    }
+
+    private static DataType resolveInputType(Collection<? extends Expression> list) {
+        DataType previousInput = null;
+        DataType previousOutput = null;
+        for (Expression choice : list) {
+            DataType thisInput = choice.requiredInputType();
+            if (previousInput == null)
+                previousInput = thisInput;
+            else if (thisInput != null && !previousInput.isAssignableFrom(thisInput))
+                throw new VerificationException(ScriptExpression.class, "Choice expressions require conflicting input types, " +
```

Comment:
Should the exception message be "require non-conflicting input types" instead?

Reply:
done